packages/HorizonUI/Sources/HorizonUI/Sources/Components/Tag/HorizonUI.Tag.swift

```diff
@@ -0,0 +1,214 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+public extension HorizonUI {
+    struct Tag: View {
+        public enum Size {
+            case small
+            case medium
+            case large
+
+            var typography: Typography.Name {
+                switch self {
+                case .small: return .p3
+                case .medium: return .buttonTextMedium
+                case .large: return .buttonTextLarge
+                }
+            }
+        }
+
+        public enum Style {
+            case standalone
+            case inline
+
+            var cornerRadius: CornerRadius {
+                switch self {
+                case .standalone:
+                    return .level5
+                case .inline:
+                    return .level1
+                }
+            }
+        }
+
+        // MARK: - Dependencies
+
+        private let title: String
+        private let style: Style
+        private let size: Size
+        private let backgroundColor: Color
+        private let borderColor: Color
+        private let onCloseAction: (() -> Void)?
+
+        // MARK: - Init
+
+        init(
+            title: String,
+            style: Style,
+            size: Size,
+            backgroundColor: Color = .huiColors.surface.pageSecondary,
+            borderColor: Color = .huiColors.lineAndBorders.lineStroke,
```

Comment:
The `backgroundColor` is added but not used anywhere, & i recommend to pass text color too as a parameter

Reply:
Good catch. I added text color and also disabled state was missing.